src/features/Chat/Thread/reducer.ts

```diff
@@ -182,4 +191,28 @@ export const chatReducer = createReducer(initialState, (builder) => {
     state.thread.title = action.payload.title;
     state.thread.isTitleGenerated = action.payload.isTitleGenerated;
   });
+
+  builder.addCase(newIntegrationChat, (state, action) => {
+    // TODO: find out about tool use
+    const next = createInitialState("agent", action.payload.integration);
+    next.thread.integration = action.payload.integration;
+    next.thread.messages = action.payload.messages;
+
```

Comment:
Delete these comments

Reply:
`TODO: find out about tool use` this one also?

Reply:
no, I think that's still up in the air with the new chat_mode object in the meta property for chat requests.